File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const map = new Map();
    
    for(let i =0; i< participant.length; i++){
        let par = participant[i];
        let com = completion[i];
        
        map.set(par, (map.get(par) || 0) + 1);
        map.set(com,(map.get(com) || 0) - 1);
    }
    
    for([k,v] of map){
        if(v>0)return k;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    d = {}
    
    for i in participant:
        d[i] = d.get(i, 0) + 1
        
    for i in completion:
        if i in d:
            d[i] = d.get(i, 0) - 1
            
    for k,v in d.items():
        if v != 0:
            return k
    